watchdog/watchdog.go

```diff
@@ -90,6 +90,12 @@ func execute(endpoint *core.Endpoint, alertingConfig *alerting.Config, maintenan
 
 // UpdateEndpointStatuses updates the slice of endpoint statuses
 func UpdateEndpointStatuses(endpoint *core.Endpoint, result *core.Result) {
+	// If user has set the failed-interval field, update enpoint interval due to its previous failure
+	if !endpoint.IntervalChanged && !result.Success && endpoint.FailedInterval.Seconds() > 0 {
+		endpoint.Interval = endpoint.FailedInterval
```

Comment:
doesn't this value need to be changed back to the original value when the check is successful again?

Reply:
@heitorPB I am not sure about this part, I thought the issue owner want it to be shorten forever?

Reply:
> @heitorPB I am not sure about this part, I thought the issue owner want it to be shorten forever?

If someone wants to shorten the interval between checks forever, why change the code instead of the configuration file?

What I understood about this feature request is to have shorter intervals (more checks) _during an outage_, and after the system is healthy again go back to the regular intervals.

Does that make sense?